lib/Frontend/CachingUtils.cpp

```diff
@@ -188,6 +189,14 @@ bool replayCachedCompilerOutputs(
           if (!Proxy)
             return Proxy.takeError();
 
+          if (Kind == file_types::ID::TY_Object && UseCASBackend) {
+            auto Schema =
+                std::make_unique<llvm::mccasformats::v1::MCSchema>(CAS);
+            llvm::raw_svector_ostream OS(ContentsStorage);
+            if (auto E = Schema->serializeObjectFile(*Proxy, OS))
+              llvm::report_fatal_error(std::move(E));
```

Comment:
Again, please never use `report_fatal_error` except test utility. Just return the error.